### ISSUE TITLE
$default-animation-duration not being used?

### ISSUE THREAD

**Issue by mattejames** [Compass/compass#1556]
I recently updated to Compass version 1.0.16 and am setting up basic use of the new animation stuff.  For some reason, when I try to set the default values for different animation settings, they don't take effect, requiring that I hard-code the values throughout my app.

Specifically:

```
$default-animation-duration: 0.5s;

@import "compass/css3";

@include keyframes(slideOutLeft) {
  0% {
    @include transform(translateX(0%));
  }

  100% {
    @include transform(translateX(-100%));
  }
}

#id {
@include animation(slideOutLeft); // Doesn't work
}

#id2 {
@include animation(slideOutLeft 0.5s); // Does work.
}
```

Any thoughts?

**Comment by mirisuzanne** [comment]
Yeah, currently the defaults are only used in the long-form properties e.g. `animation-duration()`, or in when no arguments other are passed e.g. `animation()` (where all the defaults are used). Not sure that's the best way, but the defaults are pretty invasive otherwise.

**Comment by mattejames** [comment]
So is my work around to do this?:

$default-animation-duration: 0.5s;
@include animation-name(slideOutLeft)

Or is the only work-around to use long-form?

Might be misunderstanding you too: if I'm using animate-delay() as a mix-in, wouldn't the implication be that I want to override the default?  I'm new to the use of the default variables in compass, so it could be a misunderstanding about their purpose, but I took it to mean "use this as a default whenever you do anything and if I want to override it, I'll do so by manually using the appropriate mix-in or property manually".

**Comment by mirisuzanne** [comment]
Basically, the defaults are only used _when no other value is passed_.

``` scss
// no value passed, so we use all the defaults:
@include animation();

// value passed, so no defaults are used:
@include animation(slideOutLeft);

// a workaround for your case:
@include animation-name(slideOutLeft);
@include animation-duration(); // no value, so we use the default...
```

Your proposed workaround would not be effective, because the default duration is never output to CSS. I imagine we could improve the entire system of defaults now, using Sass 3.3 maps, but it would take some time to make a smooth change.

**Comment by chriseppstein** [comment]
I'm starting to wonder whether the defaults add much value at all.

**Comment by mirisuzanne** [comment]
I agree. I think they should go away. At this point, it is easier and more useful in most cases for users to be creating their own shortcuts.
